model/state_model.jl

```diff
@@ -11,418 +10,434 @@ using JuMP
+@NLparameter(cge, cd0[r in set[:r], g in set[:g]] == sld[:cd0][r,g]);
+@NLparameter(cge, c0[r in set[:r]] == sld[:c0][r]);
+@NLparameter(cge, yh0[r in set[:r], g in set[:g]] == sld[:yh0][r,g]);
+@NLparameter(cge, bopdef0[r in set[:r]] == sld[:bopdef0][r]);
+@NLparameter(cge, hhadj[r in set[:r]] == sld[:hhadj][r]);
+@NLparameter(cge, g0[r in set[:r], g in set[:g]] == sld[:g0][r,g]);
+@NLparameter(cge, xn0[r in set[:r], g in set[:g]] == sld[:xn0][r,g]);
+@NLparameter(cge, xd0[r in set[:r], g in set[:g]] == sld[:xd0][r,g]);
+@NLparameter(cge, dd0[r in set[:r], g in set[:g]] == sld[:dd0][r,g]);
+@NLparameter(cge, nd0[r in set[:r], g in set[:g]] == sld[:nd0][r,g]);
+@NLparameter(cge, i0[r in set[:r], g in set[:g]] == sld[:i0][r,g]);
+
+# benchmark value share parameters
+@NLparameter(cge, alpha_kl[r in set[:r], s in set[:s]] == ensurefinite(value(ld0[r,s]) / (value(ld0[r,s]) + value(kd0[r,s])))); 
+@NLparameter(cge, alpha_x[r in set[:r], g in set[:g]] == ensurefinite((value(x0[r,g]) - value(rx0[r,g])) / value(s0[r,g])));
+@NLparameter(cge, alpha_d[r in set[:r], g in set[:g]] == ensurefinite(value(xd0[r,g]) / value(s0[r,g])));
+@NLparameter(cge, alpha_n[r in set[:r], g in set[:g]] == ensurefinite(value(xn0[r,g]) / value(s0[r,g])));
+@NLparameter(cge, theta_n[r in set[:r], g in set[:g]] == ensurefinite(value(nd0[r,g]) / (value(nd0[r,g]) - value(dd0[r,g]))));
+@NLparameter(cge, theta_m[r in set[:r], g in set[:g]] == ensurefinite((1+value(tm0[r,g])) * value(m0[r,g]) / (value(nd0[r,g]) + value(dd0[r,g]) + (1 + value(tm0[r,g])) * value(m0[r,g]))));
+
+#Substitution and transformation elasticities
+@NLparameter(cge, es_va[r in set[:r], s in set[:s]] == SUB_ELAST[:va]); # value-added nest - substitution elasticity
+@NLparameter(cge, es_y[r in set[:r], s in set[:s]]  == SUB_ELAST[:y]); # Top-level Y nest (VA,M) - substitution elasticity
+@NLparameter(cge, es_m[r in set[:r], s in set[:s]]  == SUB_ELAST[:m]); # Materials nest - substitution elasticity
+@NLparameter(cge, et_x[r in set[:r], g in set[:g]]    == TRANS_ELAST[:x]); # Disposition, distribute regional supply to local, national, export - transformation elasticity
+@NLparameter(cge, es_a[r in set[:r], g in set[:g]]    == SUB_ELAST[:a]); # Top-level A nest for aggregate demand (set[:m], set[:g]) - substitution elasticity
+@NLparameter(cge, es_mar[r in set[:r], g in set[:g]]  == SUB_ELAST[:mar]); # Margin supply - substitution elasticity
+@NLparameter(cge, es_d[r in set[:r], g in set[:g]]    == SUB_ELAST[:d]); # Domestic demand aggregation nest (intranational) - substitution elasticity
+@NLparameter(cge, es_f[r in set[:r], g in set[:g]]    == SUB_ELAST[:f]); # Domestic and foreign demand aggregation nest (international) - substitution elasticity
 
 
 ################
 # VARIABLES
 ################
 
-cge = MCPModel();
-
-# small value that acts as a lower limit to variable values
-# default is zero
-sv = 0.00
+# Set lower bound
+lo = MODEL_LOWER_BOUND
```

Comment:
This is something that can come in handy as a lever to tweak when the model won't solve. Typically this will be set to a value of 0 and solve just fine. The recursive model won't solve if this is set to 0 though. Starting with zero or 1e-5, might be a better choice, then if I run into trouble I can just overwrite `lo` within the model.